pkg/xds/bootstrap/template_v3.go

```diff
@@ -352,7 +352,7 @@ func genConfig(parameters configParameters, proxyConfig xds.Proxy, enableReloada
 		res.Node.Metadata.Fields[core_xds.FieldDataplaneReadinessPort] = util_proto.MustNewValueForStruct(strconv.Itoa(int(parameters.ReadinessPort)))
 	}
 	// a request from an old DP will not include this field, so defaults to false
-	if parameters.AppProbeProxyEnabled {
+	if !parameters.AppProbeProxyDisabled {
```

Comment:
Look at the comment above - doesn't it mean that during upgrade old proxies will always send `false` here so it will by default be `true`?

Reply:
Yeah and no, thanks for pointing this out. 

This code [was introduced in 2.9](https://github.com/kumahq/kuma/commit/002e301acf53484f3508a2a7e4ebce8e7df97f66), at that time the CP needs to keep the compatiablity with DPs of versions 2.7 and 2.8. Now we are on a much higher version and [we are always setting this annotation](https://github.com/kumahq/kuma/blob/4479a75c816f16645d48e8d21336756783e57803/pkg/plugins/runtime/k8s/probes/pod_probe_overrider.go) since 2.9. 

So technically we can do this change now.

**Background**

The field was initially called `appProbeProxyDisabled`, and it was then renamed to `appProbeProxyEnabled`, but unfortunately I forgot updating the json tag.